OpenRA.Mods.Common/Activities/Air/HeliReturnToBase.cs

```diff
@@ -46,13 +46,28 @@ public override Activity Tick(Actor self)
 									.Where(a => a.Owner == self.Owner && rearmBuildings.Contains(a.Info.Name))
 									.ClosestTo(self);
 
-				if (nearestHpad == null)
+				if (nearestHpad == null) {
 					return ActivityUtils.SequenceActivities(new Turn(self, initialFacing), new HeliLand(self, true), NextActivity);
-				else
-					return ActivityUtils.SequenceActivities(new HeliFly(self, Target.FromActor(nearestHpad)));
+				} else {
+					var distanceFromHelipad = (nearestHpad.CenterPosition - self.CenterPosition).HorizontalLength;
+					var distanceLenght = heli.Info.WaitDistanceFromHelipad.Length;
+
+					// If no pad is avaliable, move near one and wait
+					if (distanceFromHelipad > distanceLenght) {
```

Comment:
While you're at it, `Lenght` should be `Length`, and `available` is mistyped as well.